resources/schema/hdb_schema.sql

```diff
@@ -667,6 +671,22 @@ CREATE INDEX IF NOT EXISTS att_scalar_devenum_att_conf_id_idx ON att_scalar_deve
 CREATE INDEX IF NOT EXISTS att_scalar_devenum_att_conf_id_data_time_idx ON att_scalar_devenum (att_conf_id,data_time DESC);
 SELECT create_hypertable('att_scalar_devenum', 'data_time', chunk_time_interval => interval '28 day', create_default_indexes => FALSE);
 
+-- Trigger to set the enum_labels
+CREATE OR REPLACE FUNCTION set_enum_label() RETURNS TRIGGER AS $$
+DECLARE
+BEGIN
+    IF NEW.value_r IS NOT NULL THEN
+        NEW.value_r_label := (SELECT enum_labels[NEW.value_r + 1] FROM att_parameter WHERE att_conf_id=NEW.att_conf_id ORDER BY recv_time DESC LIMIT 1);
```

Comment:
It should not happen,  but what happens if enum_labels[NEW.value_r+1] does not exist in att_parameter?
For instance, a problem occurred when an AttributeConfiguration event was received and the enum_labels could not be saved in att_parameter DB table.
Is this `SELECT enum_labels[NEW.value_r + 1] FROM att_parameter WHERE att_conf_id=NEW.att_conf_id ORDER BY recv_time DESC LIMIT 1)` returning NULL in this case or is there some kind of error raised?
If there is an error, would this completely cancel the current transaction?

Reply:
To be tested indeed, but I believe the select statement will just return an empty row, and the value_r_label should be null.